github/event_types.go

```diff
@@ -459,14 +459,17 @@ type StatusEvent struct {
 	Branches    []*Branch `json:"branches,omitempty"`
 
 	// The following fields are only populated by Webhook events.
-	ID        *int             `json:"id,omitempty"`
-	Name      *string          `json:"name,omitempty"`
-	Context   *string          `json:"context,omitempty"`
-	Commit    *PushEventCommit `json:"commit,omitempty"`
-	CreatedAt *Timestamp       `json:"created_at,omitempty"`
-	UpdatedAt *Timestamp       `json:"updated_at,omitempty"`
-	Repo      *Repository      `json:"repository,omitempty"`
-	Sender    *User            `json:"sender,omitempty"`
+	ID      *int    `json:"id,omitempty"`
+	Name    *string `json:"name,omitempty"`
+	Context *string `json:"context,omitempty"`
+	Commit  *struct {
+		SHA    *string          `json:"sha,omitempty"`
+		Commit *PushEventCommit `json:"commit,omitempty"`
+	}
```

Comment:
[`RepositoryCommit` type](https://godoc.org/github.com/google/go-github/github#RepositoryCommit) looks like a close fit, but it's missing `url` and `comments_url`. It contains [`Commit` type](https://godoc.org/github.com/google/go-github/github#Commit) which is a close fit for the inner commit.

So perhaps we can use that (and add missing fields to it), or a custom new type(s) just for this event.